### PR TITLE
Keep track of generated files to enable cleanup

### PR DESCRIPTION
**Pull request by rlmartin**
This is useful when regenerating code. If the file list has changed,
`apibuilder update` will be able to remove those files automatically.
This feature is opt-in (via a setting in config), so existing usage
will not be affected.

### CODE REVIEW

File: src/apibuilder-cli/file_tracker.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+module ApibuilderCli
```
Comment (mbryzek) [inline src/apibuilder-cli/file_tracker.rb]: possible to indent with two spaces?

File: src/apibuilder-cli/file_tracker.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  class FileTracker
+
+  	def FileTracker.default_path
+  		"#{ApibuilderCli::Config::APIBUILDER_LOCAL_DIR}/tracked_files"
```
Comment (mbryzek) [inline src/apibuilder-cli/file_tracker.rb]: should we make this `.tracked_files` ?

Comment (rlmartin) [reply src/apibuilder-cli/file_tracker.rb]: I was thinking about that. I was taking a page from the git files, where top-level files/directories are '.' prefixed (`.gitignore`, `.git` dir), but files inside those aren't (`.git/config`). The main concern I was thinking about is that there might be a name collision with scaffolding source files that are dropped into the `.apibuilder` directory (i.e. if a generator produces a scaffolding file named `tracked_files`). So this is a decent argument for the prefix.

**Comment by rlmartin** [comment]
I actually think the opposite is necessary - that this file _must_ be checked into git. The purpose is to keep track of the apibuilder-generated files across code-generation events - it is not something used within a single generation cycle. So if you generate the code, commit it, then manually add some things, then regenerate the code (perhaps on a different dev machine), this file needs to be present across all of that. Seeing this file committed into code also makes it explicit which files are "owned" by apibuilder and which aren't. This could make obsolete the need to sequester files in `generated` subdirectories.

File: src/apibuilder-cli/file_tracker.rb (hunk 1 above)
Comment (rlmartin) [reply src/apibuilder-cli/file_tracker.rb]: Definitely. Not sure why this happened - my IDE (SublimeText) is visually showing this as two spaces, not 4. I'll fix this.

**Comment by mbryzek** [comment]
ack - +1

**Pull request merged**